ibis/backends/tests/test_generic.py

```diff
@@ -1436,6 +1437,53 @@ def test_pivot_wider(backend):
     assert len(df) == diamonds.color.nunique().execute()
 
 
+def test_select_distinct_order_by_alias(backend, con):
+    df = pd.DataFrame({"x": [1, 2, 3, 3], "y": [10, 9, 8, 8]})
+    expr = ibis.memtable(df).select(y="x", x="y").distinct().order_by("x", "y")
+    sol = (
+        df.drop_duplicates()
+        .rename(columns={"x": "y", "y": "x"})
+        .sort_values(["x", "y"])
+    )
+    res = con.to_pandas(expr)
+    backend.assert_frame_equal(res, sol)
+
+
+def test_select_distinct_order_by_expr(backend, alltypes, df):
+    res = alltypes.select("int_col").distinct().order_by(-_.int_col).to_pandas()
+    sol = df[["int_col"]].drop_duplicates().sort_values("int_col", ascending=False)
+    backend.assert_frame_equal(res, sol)
+
+
+@pytest.mark.parametrize(
+    "ops",
+    [
+        param(ops, id="-".join(ops))
+        for ops in permutations(("select", "distinct", "filter", "order_by"))
+        if ops.index("select") < ops.index("distinct")
+    ],
+)
+def test_select_distinct_filter_order_by_commute(backend, alltypes, df, ops):
```

Comment:
When fusing is enabled (which it is by default), some of our operations are commutative (or at least should be).

- `select`/`filter`/`order_by` always commute (as well as their variants like `drop`).
- `distinct` also commutes with these (sans `select` in most cases), provided certain restrictions are met. A little more nuanced.

This test checks that behavior across our backends so users can rely on it a little more. This turned up bugs.

Reply:
Do we need some issues, or are these fixed in this PR?

Reply:
See https://github.com/ibis-project/ibis/issues/10005